strawberryfields/api/connection.py

```diff
@@ -276,12 +276,31 @@ def get_job_result(self, job_id: str) -> Result:
                 buf.seek(0)
 
                 samples = np.load(buf, allow_pickle=False)
+                if isinstance(samples, np.lib.npyio.NpzFile):
+                    samples = dict(samples)  # convert to dict to free buffer
 
+            # Samples represent photon numbers.
+            # Convert to int64, to avoid unexpected behaviour
+            # when users postprocess these samples.
+            def int_to_int64(samples):
                 if np.issubdtype(samples.dtype, np.integer):
-                    # Samples represent photon numbers.
-                    # Convert to int64, to avoid unexpected behaviour
-                    # when users postprocess these samples.
-                    samples = samples.astype(np.int64)
+                    return samples.astype(np.int64)
+                return samples
+
+            if isinstance(samples, dict):
+                samples = {key: int_to_int64(value) for key, value in samples.items()}
+            else:
+                samples = int_to_int64(samples)
+
+            # The npz format can be used to compress data. In this case samples is
+            # a dict at this point, but the user might expect a single array. For
+            # this case we use the special key ``single_samples_array``.
+            if (
+                isinstance(samples, dict)
+                and "_single_samples_array" in samples
+                and len(samples) == 1
+            ):
+                samples = samples["_single_samples_array"]
```

Comment:
this choice of convention is arbitrary, and maybe not the best one. If we went with this kind of pattern to transport single arrays in npz format, where would we best document it?

Reply:
Are you sure we need to establish such a convention? If we know that each SF job produces exactly one output (i.e., an array of samples), would it not be more appropriate to use
```python
samples = next(samples.values())
```
in the event where `samples` is a dictionary?  We can raise an exception if the dictionary has more than one value.

Reply:
what's wrong with "samples"? would this conflict with the keys used by QKD results?